src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -63,12 +64,15 @@ public function process(ContainerBuilder $container)
         $definition = $container->findDefinition($this->dispatcherService);
 
         foreach ($container->findTaggedServiceIds($this->listenerTag, true) as $id => $events) {
+            $isAutoregisterEnabled = false;
+            $registeredMethods = [];
             foreach ($events as $event) {
-                $priority = isset($event['priority']) ? $event['priority'] : 0;
-
                 if (!isset($event['event'])) {
-                    throw new InvalidArgumentException(sprintf('Service "%s" must define the "event" attribute on "%s" tags.', $id, $this->listenerTag));
```

Comment:
I would require one of: the attribute is present *or* the class implements EventDispatcherInterface (the interface check shouldn't be done when the attribute is set)

Reply:
done. Delayed in `if ($isAutoregisterEnabled)` to avoid unecessary reflexion